lib/fhir_client/client.rb

```diff
@@ -279,22 +265,31 @@ def try_conformance_formats(default_format)
       formats.insert(0, default_format)
 
       @cached_capability_statement = nil
-      @default_format = nil
 
       formats.each do |frmt|
         reply = get 'metadata', fhir_headers({accept: "#{frmt}"})
         next unless reply.code == 200
+        use_r4
         begin
           @cached_capability_statement = parse_reply(FHIR::CapabilityStatement, frmt, reply)
         rescue
           @cached_capability_statement = nil
         end
-        unless @cached_capability_statement
+        if @cached_capability_statement.nil? || !@cached_capability_statement.fhirVersion.starts_with?('4')
+          use_stu3
           begin
-            @cached_capability_statement = parse_reply(FHIR::DSTU2::Conformance, frmt, reply)
+            @cached_capability_statement = parse_reply(FHIR::STU3::CapabilityStatement, frmt, reply)
           rescue
             @cached_capability_statement = nil
           end
+          use_dstu2
```

Comment:
I think this `use_dstu2` should be inside the block under the next line `unless @cached_capability_statement` yeah?

Reply:
Yeah.  Just moved it, thanks for catching that.